frontend/class-opengraph-oembed.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Frontend
+ */
+
+/**
+ * Class WPSEO_OpenGraph_OEmbed
+ */
+class WPSEO_OpenGraph_OEmbed implements WPSEO_WordPress_Integration {
+	/**
+	 * Registers the hooks.
+	 */
+	public function register_hooks() {
+		add_filter( 'oembed_response_data', array( $this, 'wpseo_hook_oembed' ), 10, 4 );
```

Comment:
It's possible to check if OpenGraph is enabled before `add_filter` to improve performance.
```php
public function register_hooks() {
		if ( ! WPSEO_Options::get( 'opengraph' ) ) {
		   return;
        }
```

Reply:
I would assume that if I have this check then I dont need the other one. I will move the other check to here.

Reply:
Also for this update is there a reason you prefer to to a `! opengraph` check instead of just wrapping the add_filter in a `if( opengraph )` (pardon the pseudo-code)?

Reply:
Yes, we don't need check in the method `wpseo_hook_oembed` if we move it earlier. Ideally we should check it before instantiation of the class (same as for `WPSEO_OpenGraph`);
I think that's common way, but we should wait for other feedback. I just want to point to possible improvement.

Reply:
👍 I am happy to update if needed. Thanks!